python/CCLE_data.py

```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
 
 #Used to create correlation matrix
 def Matrix(gene_list, histone_list, data1, data2):
-    name = (len(gene_list), len(histone_list))
-    name = np.zeros(name)
+    """This functions purpose is to create a numpy matrix which is filled with 
+    correlation values based on two data sets and corresponding lists"""
+    Rmat = (len(gene_list), len(histone_list))
+    Rmat = np.zeros(Rmat)
+    Pmat = Rmat
     i = 0 
     for gene in gene_list:
         j = 0
         for histone in histone_list:
             data1.loc[gene] = data1.loc[gene].fillna(method='ffill')
             data2.loc[histone] = data2.loc[histone].fillna(method='ffill')
             correlation = pearsonr(data1.loc[gene], data2.loc[histone])
-            correlation = correlation[0]
-            name[i][j] = correlation
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
             j = j+1
         i = i+1
-    return name
+    return Rmat, Pmat
 
-#Used to make heatmaps
-def Heatmap(data, size, colour, title):
+def Heatmap(data, ylabels, size, colour, title):
+    """Using dataframes based on the Matrix fucntion, this function will create 
+    a heatmap"""
     plt.figure(figsize=size)
     plt.xlabel('Histone Markers')
     plt.title(title)
-    ax = sns.heatmap(data, cmap = colour)
+    ax = sns.heatmap(data, cmap = colour, square = True, yticklabels = ylabels)
     ax.set(xlabel = 'Histone Markers')
     plt.show
+    
+def pearson_dfs(gene_list, histone_list, data1, data2):
+    """This functions purpose is to create two dataframes. The first dataframe will contain all the r-values between the two sets of data. The second will contain all the p-values for each r-value.
+    
+    This function is very specific. The dataframe created will contain histone markers as the columns and gene names as the index. 
+    
+    This function assumes you already took the intersection between the two arrays. If you do not do that, then the matrix dimensions will not agree, resulting in an error.
+    
+    The format for inputting matrix should have columns correspond to the celllines.
+    """
+    Rmat = (len(gene_list), len(histone_list))
+    Rmat = np.zeros(Rmat)
+    Pmat = Rmat[:]
+    i = 0 
+    for gene in gene_list:
+        j = 0
+        for histone in histone_list:
+            data1.loc[gene] = data1.loc[gene].fillna(method='ffill')
+            data2.loc[histone] = data2.loc[histone].fillna(method='ffill')
+            correlation = pearsonr(data1.loc[gene], data2.loc[histone])
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
+            j = j+1
+        i = i+1
+        
+    Rdf = pd.DataFrame(Rmat)
```

Comment:
It looks like you can merge Matrix and pearson_dfs into a single function. This seems redundant.